frontend/src/components/ProfileContent.tsx

```diff
@@ -50,64 +62,160 @@ const useStyles = makeStyles({
         padding: 5,
         paddingRight: 10,
     },
-});
+    bottomRight: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-10%, -10%)',
+    },
+    bottomRight2: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-100%, -10%)',
+    },
+})
 
 export default function ProfileContent(props: ProfileContentProps) {
-    const classes = useStyles();
+    const classes = useStyles()
+    const [isEditMode, toggleEditMode] = React.useState(false)
+
+    // Pass props to to useProfile hook.
+    const { editableContent, setters } = useProfile(props)
+
+    // Initialize readonly profile content and acquire hook to update it when edits are saved.
+    const [readonlyContent, setProfile] = React.useState<ProfileContentProps>(editableContent);
 
-    // Add user profile values to list.
-    var profileFields: ProfileField[] = [
-        {
+    // User profile field list. Field values are assigned to readonly content.
+    const fields: ProfileFieldList = {
+        name: {
             title: 'Full Name',
-            value: props.name
+            value: readonlyContent.name,
         },
-        {
+        username: {
             title: 'Username',
-            value: props.username
+            value: readonlyContent.username,
         },
-        {
+        age: {
             title: 'Age',
-            value: props.age.toString()
+            value: readonlyContent.age.toString(),
         },
-        {
+        gender: {
             title: 'Gender',
-            value: props.gender
+            value: readonlyContent.gender,
         },
-        {
+        occupations: {
             title: 'Occupations',
-            value: props.occupations.join(', ')
+            value: readonlyContent.occupations.join(', '),
         },
-        {
+        location: {
             title: 'Location',
-            value: props.location
+            value: readonlyContent.location,
         },
-        {
+        interests: {
             title: 'Interests',
-            value: props.interests.join(', ')
+            value: readonlyContent.interests.join(', '),
         },
-        {
+        religion: {
```

Comment:
It strikes me that if people can have multiple occupations, they can have multiple religions. Just a thought. It's perfectly compatible to be Jewish and Atheist. Or Unitarian and Buddhist.

Reply:
agreed